src/libworkbench/meson.build

```diff
@@ -0,0 +1,116 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+# SPDX-FileCopyrightText: Andy Holmes <andrew.g.r.holmes@gmail.com>
+
+
+prefix = get_option('prefix')
+datadir = join_paths(prefix, get_option('datadir'))
+includedir = join_paths(prefix, get_option('includedir'))
+libdir = join_paths(prefix, get_option('libdir'))
+
+girdir = join_paths(datadir, 'gir-1.0')
+typelibdir = join_paths(libdir, 'girepository-1.0')
+vapidir = join_paths(datadir, 'vala', 'vapi')
+
+libworkbench_api_version = '0'
+libworkbench_api_name = 'libworkbench-@0@'.format(libworkbench_api_version)
+
+
+#
+# Shared Library
+#
+libworkbench_c_args = [
+  '-Wall',
+  '-Wextra',
+  '-Wfloat-equal',
+  '-Wformat=2',
+  '-Wincompatible-pointer-types',
+  '-Wint-conversion',
+  '-Wint-to-pointer-cast',
+  '-Wmissing-include-dirs',
+  '-Woverflow',
+  '-Wpointer-arith',
+  '-Wpointer-to-int-cast',
+  '-Wredundant-decls',
+  '-Wshadow',
+  '-Wstrict-prototypes',
+  '-Wundef',
+
+  '-Wno-discarded-array-qualifiers',
+  '-Wno-missing-field-initializers',
+  '-Wno-unused-parameter',
+  '-Wno-missing-declarations',
+
+  '-DWORKBENCH_COMPILATION',
+  '-DWORKBENCH_EXPORT=__attribute__((visibility("default"))) extern',
+]
+libworkbench_link_args = []
+libworkbench_deps = [
+  dependency('gio-2.0', version: '>= 2.76.0'),
+  dependency('gtk4', version: '>= 4.10.0'),
+  dependency('gtksourceview-5', version: '>= 5.8.0'),
+]
+
+if get_option('profile') != 'development'
+  libworkbench_c_args += [
+    '-DG_DISABLE_ASSERT',
```

Comment:
interesting

From what I understand Gtk / GLib doesn't do that rigth?

Reply:
I believe it does based on `debug` or `optimization` meson options, but those don't really apply to Workbench I think.

`g_assert()` failures are hard crashes with line numbers, so they're useful when developing, but as a rule if they fail you made a mistake that shouldn't ever ship.